### DISCUSSION TITLE
CORS issue after upgrading [Feb 09]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#5128] [Discord Threads]
> 🙋‍♂️ **ebuyum**

**Comment by o1lab** [comment]
> 💁 **o1lab**  
#### Is it just the swagger API request ? or you can call it from say an API Client ?  
#  
> 🙋‍♂️ **ebuyum**  
#### Any API client works fine, postman for example works well  
#  
> 🙋‍♂️ **ebuyum**  
#### But, when trying to use a react app using axios library is when I fail  
#  
> 💁 **o1lab**  
#### axios has to add extra header for this   
#### could you please search  
#  
> 🙋‍♂️ **ebuyum**  
#### extra header for cors?  
#  
> 🙋‍♂️ **ebuyum**  
#### I'm checking  
#  
> 🙋‍♂️ **ebuyum**  
#### Tried adding 'Access-Control-Allow-Origin': '*' to the request headers but didn't seem to help  
#  
> 🙋‍♂️ **ebuyum**  
#### Checking online suggests using packages such as http-proxy-middleware, but its interesting that something changed during the versions  
#  
> 👨‍💻 **pranavxc**  
#### Update `index.js` file inside the generated code to enable Cross origin access   
####    
#### ```js   
#### const cors = require("cors");   
#### (async () => {   
####     const app = require('express')();   
####     const {Noco} = require("nocodb");   
####     app.use(   
####         cors({   
####           origin: "*",   
####         })   
####     );   
####     app.use(await Noco.init({}));   
####     console.log(`Visit : localhost:${process.env.PORT}/dashboard`)       
####     app.listen(process.env.PORT);   
#### })()   
#### ```  
#  
> 🙋‍♂️ **ebuyum**  
#### Ah, ok, interesting   
#### let me try, since I've got nocodb deployed on a cloud foundry instance  
#  
> 🙋‍♂️ **ebuyum**  
#### I'm guessing this isn't exposed to an environment variable?  
#  
> 👨‍💻 **pranavxc**  
#### earlier did it worked with react ?  
#  
> 🙋‍♂️ **ebuyum**  
#### Yeah  
#  
> 🙋‍♂️ **ebuyum**  
#### But this was a really old version  
#  
> 🙋‍♂️ **ebuyum**  
#### Ah, fixed it completely  
#  
> 🙋‍♂️ **ebuyum**  
#### You guys rock, thanks a lot!  
#  
> 🙋‍♂️ **ebuyum**  
#### I'm guessing the nocodb-seed project doesn't get updated a lot?   
#### I can just use a modified index.js file and still have latest nocodb right?  
#  
> 👨‍💻 **pranavxc**  
#### it's better to avoid using `*` - use the domain name you are going to use as origin, refer https://www.npmjs.com/package/cors  
#  
> 👨‍💻 **pranavxc**  
#### yes.... nocodb is installed as npm module  
#  
> 🙋‍♂️ **ebuyum**  
#### Great  
#  
> 🙋‍♂️ **ebuyum**  
#### Yeah, but this is an internal installation, so shouldn't be much of a problem I hope since I'm using nocodb as an API service mostly  
#  
> 🙋‍♂️ **Dassaev**  
#### Thank you I solved my problem with this code!  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/925384641513390121)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
